todos/src/reducers/todoReducers.js

```diff
@@ -46,7 +47,54 @@ const todoReducers = (state = initialData, action) => {
         ...state,
         list: updatedList,
       };
+    case "EDIT_TODO":
+      const updatedList2 = state.list.map((todo) => {
+        if (todo.id === action.id) {
+          return {
+            ...todo,
+            onEdit: action.onEdit,
+          };
+        }
+        return todo;
+      });
 
+      return {
+        ...state,
+        list: updatedList2,
+      };
+    case "EDIT_UPDATE_COMPLETED":
+      const editUdupdatedList = state.list.map((todo) => {
+        if (todo.id === action.id) {
+          return {
+            ...todo,
+            isCompleted: action.isCompleted,
+            date: action.date,
+            completedDate: new Date().toLocaleDateString(),
```

Comment:
Can we send the calculated date from the component and just use it as placeholder here? This is because of the point I made earlier. Reducers must be pure.

Reply:
@hhu-wtag Bhai,
Is it okay to set my`completedDate` into my action (as I'm getting my `completedDat`e as `null` into the component that is dispatching the action) and use `completedDate: action.completedDate,` inside the reducer to keep it clean as pure function?

Reply:
Yes, you can do asynchronous tasks in actions and send it to the reducers. Just remember always to keep the reducers side effect free.